src/util/bip32.rs

```diff
@@ -504,6 +504,21 @@ impl ExtendedPrivKey {
         })
     }
 
+    /// Constructs ECDSA compressed private key matching internal secret key representation.
+    pub fn to_ecdsa(&self) -> ecdsa::PrivateKey {
+        ecdsa::PrivateKey {
+            compressed: true,
+            network: self.network,
+            key: self.private_key
+        }
+    }
+
+    /// Constructs BIP340 keypair for Schnorr signatures and Taproot use matching the internal
+    /// secret key representation.
+    pub fn to_schnorr<C: secp256k1::Signing>(&self, secp: &Secp256k1<C>) -> schnorr::KeyPair {
```

Comment:
I have a slight preference in favor of `to_schnorr_keypair`. 

I also like `to_ecdsa_privkey`, `to_ecdsa_pubkey`, and `to_schnorr_pubkey` over the current ones. But this one is more debatable since the user has the context of what return type to expect.

Reply:
Changed to `to_keypair` since there is no other keypair (and @apoelstra does not like "Schnorr" surname :D because of him patenting signature algo)

Reply:
I would prefer `to_schnorr_keypair`. It's sad that this has patent history, but IMO the convenience and usability triumphs this.

Reply:
Do we have a keypair type for non-BIP-340 keys?

Reply:
I really do not like inconsistency in function names for extended keys that we have today in the API: `derive_priv`, `derive_pub`, `new_master`, `from_private` - and now `to_ecdsa_privkey`.... :(

Reply:
Did a temporal version with `priv/pub` instead of `privkey` for more consisted naming for now; will wait for others to comment

Reply:
Reviewing the discussion, after some thinking I think my vote is for `to_ec_pubkey`, `to_xonly_keypair` for rationale mentioned here: 
https://github.com/rust-bitcoin/rust-bitcoin/issues/588#issuecomment-890586744